Dockerfile

```diff
@@ -0,0 +1,38 @@
+# Prerequisites for using dockerfile can be seen here https://github.com/zephyrtronium/robot/issues/76
+
+# Main image
+FROM golang:1.23-alpine AS build
+
+# Define config
+ARG CONFIG
+
+# Copy repo and populate Go resources
+COPY . .
+RUN go mod download
```

Comment:
Topi's example ran `go mod download` because it copied go.mod and go.sum explicitly prior to copying the entire repo in. Doing that allows dependencies to be cached across container builds, which accelerates build time. If we aren't doing that, then `go build` later will implicitly run `go mod download`. But we should probably do that.